cmd/erasure-object.go

```diff
@@ -1177,7 +1178,7 @@ func (er erasureObjects) DeleteObject(ctx context.Context, bucket, object string
 		if disk != nil && disk.IsOnline() {
 			continue
 		}
-		er.addPartial(bucket, object, opts.VersionID)
+		er.addPartial(bucket, object, opts.VersionID, 0)
```

Comment:
Should be set to -1 to indicate delete? since fi.Size can be 0 for 0 byte objects.